Sources/System/FilePath/FilePathWindows.swift

```diff
@@ -461,3 +461,46 @@ extension SystemString {
     return lexer.current
   }
 }
+
+#if os(Windows)
+import WinSDK
+
+extension UnsafePointer where Pointee == CInterop.PlatformChar {
+  /// Invokes `body` with a resolved and potentially `\\?\`-prefixed version of the pointee,
+  /// to ensure long paths greater than 260 characters are handled correctly.
+  ///
+  /// - seealso: https://learn.microsoft.com/en-us/windows/win32/fileio/maximum-file-path-limitation
+  internal func withCanonicalPathRepresentation<Result>(_ body: (Self) throws -> Result) throws -> Result {
```

Comment:
Should `FilePath` offer this as a public operation?

Reply:
It's tempting, but I'd rather leave that to a future enhancement, I'm not sure yet that this should be a public API or what the right shape would be for it, since we'd have to commit to supporting it if we were to make that change.